src/control.class.js

```diff
@@ -217,6 +249,55 @@
       return point;
     },
 
+    /**
+     * Returns the coords for this control based on object values.
+     * @param {Number} objectAngle angle from the fabric object holding the control
+     * @param {Number} objectCornerSize cornerSize from the fabric object holding the control (or touchCornerSize if
+     *   isTouch is true)
+     * @param {Number} centerX x coordinate where the control center should be
+     * @param {Number} centerY y coordinate where the control center should be
+     * @param {boolean} isTouch true if touch corner, false if normal corner
+     */
+    calcCornerCoords: function(objectAngle, objectCornerSize, centerX, centerY, isTouch) {
+      var controlTriangleAngle,
+          cornerHypotenuse,
+          xSize = (isTouch) ? this.touchCornerSizeX : this.cornerSizeX,
+          ySize = (isTouch) ? this.touchCornerSizeY : this.cornerSizeY;
+      if (xSize && ySize) {
+        // handle custom corner sizes
+        controlTriangleAngle = Math.atan2(ySize, xSize);
+        cornerHypotenuse = Math.sqrt(xSize * xSize + ySize * ySize) / 2;
+      }
+      else {
+        // use default object corner sizes
+        controlTriangleAngle = Math.PI / 4;
+        /* 0.7071067812 stands for sqrt(2)/2 */
+        cornerHypotenuse = objectCornerSize * 0.7071067812;
+      }
+      var newTheta = controlTriangleAngle - fabric.util.degreesToRadians(objectAngle),
+          cosHalfOffset = cornerHypotenuse * fabric.util.cos(newTheta),
+          sinHalfOffset = cornerHypotenuse * fabric.util.sin(newTheta);
+
+      return {
+        tl: {
+          x: centerX - cosHalfOffset,
+          y: centerY - sinHalfOffset,
+        },
+        tr: {
+          x: centerX + cosHalfOffset,
+          y: centerY - sinHalfOffset,
+        },
+        bl: {
+          x: centerX - cosHalfOffset,
+          y: centerY + sinHalfOffset,
+        },
+        br: {
+          x: centerX + cosHalfOffset,
+          y: centerY + sinHalfOffset,
+        },
```

Comment:
this bunch of numbers here i remember they work with square and are a simplified version. Once the object is rotated i believe with a strongly rectangular controls, this does not work.
This same thing overly complicated the padding logic elsewhere.

the x and the y coordiante of each point when rotated are both affected by cos and sin of the 2 offsets.

Reply:
There should be some debug code to uncomment to see those points on the canvas.

Reply:
Yes, I think I understand the original code now. It's using the complementary angle to triangulate the other corner (which both angles are 45 when in a square, which is why stuff works but it's almost hard-coding to run the original way). Things were working with right angles when I'd rotate so thanks for challenging me to check other rotations.

I'm not sure about debug code though. I did find things below a getImageLines call that I was able to uncomment and see some squares show up so that was hepful.

Reply:
@asturur Is there anything else you'd like to see here? I tried adding padding to a shape and the rectangular corner worked, even when rotating.